datasquare.go

```diff
@@ -335,7 +335,7 @@ func (ds *dataSquare) SetCell(x uint, y uint, newShare []byte) error {
 
 // Flattened returns the concatenated rows of the data square.
 func (ds *dataSquare) Flattened() [][]byte {
-	flattened := [][]byte(nil)
+	flattened := make([][]byte, 0, ds.width*ds.width)
```

Comment:
[nit] For better readability, this portion of the code i.e., `ds.width*ds.width` could be converted to a func that calculates the number of shares in a flattened data square.